### ISSUE TITLE
Raw string literals in labels and titles 

### ISSUE THREAD

**Issue by jbdv-no** [jacobwilliams/pyplot-fortran#26]
Thank you for this incredibly useful project!

### Problem description
When labels or titles are defined that contain [escape sequences](https://docs.python.org/2.0/ref/strings.html), Python/matplotlib does not render the TeX strings correctly, e.g.:

``` fortran
call plt%initialize(..., ylabel='$\frac{\bar{x}}{\bar{x}_0}$', title='normalised $\bar{x}$ values', ...)
...
call plt%add_plot(..., label='$\bar{x}_0 \neq 0$', ...)
```
This is because e.g. the `\b`, `\f` and `\n` are interpreted by Python as escape sequences for ASCII Backspace, ASCII Formfeed and ASCII Linefeed, respectively. At least, that is the case for me with Python 3.6, running on Ubuntu 18.04  via Windows Subsystem for Linux (WSL).

### Possible solution
If the labels/titles are defined as raw string literals in Python notation, e.g. `r"$\bar{x}_0 \neq 0$"`, this problem is avoided. This can be done by changing lines [261 - 264](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L261-L264) in the `initialize` subroutine:

```fortran
if (present(xlabel)) call me%add_str('ax.set_xlabel(r"'//trim(xlabel)//'")')
if (present(ylabel)) call me%add_str('ax.set_ylabel(r"'//trim(ylabel)//'")')
if (present(zlabel)) call me%add_str('ax.set_zlabel(r"'//trim(zlabel)//'")')
if (present(title))  call me%add_str('ax.set_title(r"' //trim(title) //'")')
```
and everywhere `label` is defined in `add_***` subroutines for the different plot types (in lines [330](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L330), [411](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L411), [513](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L513), [603](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L603), [677](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L677) and [834](https://github.com/jacobwilliams/pyplot-fortran/blob/2614fbc8d14f52ddc327faa890850987e6c0a345/src/pyplot_module.f90#L834)):

```fortran
'label=r"'//trim(label)//'")'
```
There may very well be reasons for not doing this, so I'm just wondering if this is a _bug_ or a _feature_ :-)

**Comment by jacobwilliams** [comment]
I'd be OK with adding this. I can't think of a reason not to... is there any reason it might mess up some other strings?

**Comment by jacobwilliams** [comment]
You can always escape your fortran string like this `'$\\bar{x}_0 \\neq 0$'` and it would work.

I could make the raw string mode an option.

**Comment by jacobwilliams** [comment]
Fixed by #29

**Comment by jbdv-no** [comment]
Thanks a lot for the feedback and update @jacobwilliams. I am not aware of how raw strings could mess things up (although that is perhaps not saying too much...), and I think the solution you implemented making raw strings an optional setting handles all use cases perfectly. 

(Escaping the fortran strings are certainly an option, requiring no changes to the library, I just like not having to modify e.g. a label string that is used/copied from a LaTeX document elsewhere, and the new optional setting allows me to not change my old habits :-) )
